### PR TITLE
Documentation for full API in spatiotemporal.py

### PR DESCRIPTION
**Pull request by fsaad**

### CODE REVIEW

File: src/bayesnf/__init__.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 # A new PyPI release will be pushed every time `__version__` is increased.
 # When changing this, also update the CHANGELOG.md
```
Comment (ColCarroll) [inline src/bayesnf/__init__.py:18]: do you want to keep the CHANGELOG? You could update that if so, or delete the file and this note if not!

File: src/bayesnf/spatiotemporal.py [hunk 2]
```diff
@@ -288,10 +390,42 @@ def predict(self, table, quantiles=(0.5,), approximate_quantiles=False):
...
+
+        - The columns of `table` should contain all `self.feature_cols`
+          and the `self.target_col`.
+
+        - The type of the "time" column (i.e., `self.feature_cols[0]`)
+          should be `datetime`. To ensure this requirement holds, see
+          [`pandas.to_datetime`](https://pandas.pydata.org/docs/reference/api/pandas.to_datetime.html).
+          The types of the remaining feature columns should be numeric.
+
+      seed (jax.random.PRNGKey): The jax random key.
```
Comment (brianwa84) [inline src/bayesnf/spatiotemporal.py:409]: technically this is a jax.Array with dtype jax.dtypes.prng_key

**Review by brianwa84 — APPROVED** [review]

File: src/bayesnf/spatiotemporal.py [hunk 3]
```diff
@@ -186,27 +187,103 @@ def get_input_scales(self) -> np.ndarray:
...
       feature_cols: Sequence[str],
       target_col: str,
-      timetype: str,
-      freq: str,
       seasonality_periods: Sequence[float | str] | None = None,
       num_seasonal_harmonics: Sequence[int] | None = None,
       fourier_degrees: Sequence[float] | None = None,
-      observation_model: str = 'NORMAL',
+      interactions: Sequence[tuple[int, int]] | None = None,
+      freq: str,
```
Comment (ColCarroll) [inline src/bayesnf/spatiotemporal.py:219]: needs a default, or to be listed earlier

Comment (fsaad) [reply src/bayesnf/spatiotemporal.py:219]: The signature starts with `self, *, ...`, which treats `freq` as a "required" kwarg.Does it matter if there is a `*`?

Comment (ColCarroll) [reply src/bayesnf/spatiotemporal.py:219]: wild. i knew about `*`, but not that you could interleave defaults and non-defaults. I think a user would still expect to see the "required" arguments first in a signature (but after the `*`), but up to you.

File: src/bayesnf/__init__.py (hunk 1 above)
Comment (fsaad) [reply src/bayesnf/__init__.py]: Removed now.

**Pull request merged**